src/Functions/jsonMerge.cpp

```diff
@@ -0,0 +1,117 @@
+#include "config.h"
+
+#if USE_RAPIDJSON
+
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <Interpreters/Context.h>
+#include <IO/ReadBufferFromString.h>
+#include <Common/FieldVisitorToString.h>
+
+#include "rapidjson/document.h"
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+#include "rapidjson/filewritestream.h"
+#include "rapidjson/prettywriter.h"
+#include "rapidjson/filereadstream.h"
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_JSON_OBJECT_FORMAT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+    // select jsonMerge('{"a":1}','{"name": "joey"}','{"name": "tom"}','{"name": "zoey"}');
+    //            ||
+    //            \/
+    // ┌───────────────────────┐
+    // │ {"a":1,"name":"zoey"} │
+    // └───────────────────────┘
+    class FunctionjsonMerge : public IFunction
+    {
+    public:
+        static constexpr auto name = "jsonMerge";
+        static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionjsonMerge>(); }
+
+        String getName() const override { return name; }
+
+        bool isVariadic() const override { return true; }
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        size_t getNumberOfArguments() const override { return 0; }
+        bool useDefaultImplementationForConstants() const override { return true; }
+
+        DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+        {
+            if (arguments.empty())
+                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Function {} requires at least one argument.", getName());
+
+            return std::make_shared<DataTypeString>();
+        }
+
+        ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
```

Comment:
I'm not sure about what `input_rows_count` means. Can someone help me explain it?